composer/loggers/mosaicml_logger.py

```diff
@@ -80,21 +83,69 @@ def __init__(
                 log.warning(f'Environment variable `{RUN_NAME_ENV_VAR}` not set, so MosaicMLLogger '
                             'is disabled as it is unable to identify which run to log to.')
                 self._enabled = False
-
+            
     def log_hyperparameters(self, hyperparameters: Dict[str, Any]):
         self._log_metadata(hyperparameters)
 
     def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> None:
         self._log_metadata(metrics)
 
     def after_load(self, state: State, logger: Logger) -> None:
+        # Log model data downloaded and initialized for run events
+        self._log_metadata({'model_initialized_time': time.time()})
         # Log WandB run URL if it exists. Must run on after_load as WandB is setup on event init
         for callback in state.callbacks:
             if isinstance(callback, WandBLogger):
                 run_url = callback.run_url
                 if run_url is not None:
                     self._log_metadata({'wandb/run_url': run_url})
 
+    def fit_start(self, state: State, logger: Logger) -> None:
+        # adds state to calculate training progress on log metadata
+        self._state = state
```

Comment:
instead of using `self._state` to track `state` we should just pass it along on L95.